processor/k8sprocessor/kube/client.go

```diff
@@ -386,3 +476,29 @@ func selectorsFromFilters(filters Filters) (labels.Selector, fields.Selector, er
 	}
 	return labelSelector, fields.AndSelectors(selectors...), nil
 }
+
+func (c *WatchClient) addOrUpdateNamespace(namespace *api_v1.Namespace) {
+	newNamespace := &Namespace{
+		Name:         namespace.Name,
+		NamespaceUID: string(namespace.UID),
+		StartTime:    namespace.GetCreationTimestamp(),
+	}
+
+	newNamespace.Attributes = c.extractNamespaceAttributes(namespace)
+
+	c.m.Lock()
+	defer c.m.Unlock()
```

Comment:
Looks like we don't need to use defer here since the logic to be protected is quite simple and there is no risk of escaping a lock with an early return. May be guard the map assignment inside a lock/unlock pair directly?